[DOC-2229] feat(gsql): Doc for RBAC: Fine-grained query privilege: User Privilege Management; (#469)
tigergraph/server-docs: modules/user-access/pages/user-management.adoc

The above command will revoke roles `role1` and `role2` on graph `example_graph` from users `user1` and `user2`.

== Grant privileges to a user

> Users with the `WRITE_USER` privileges on global can grant RBAC privileges to the users.

Reviewer: Need to clarify better: Only users with the WRITE_USER privilege at the global level can grant RBAC privileges to other users.

Author: Ditto.

> Users with the `WRITE_USER` privileges on global can grant RBAC privileges to the users.

Reviewer 2: ```suggestion
Only users with the WRITE_USER privilege at the global level can grant RBAC privileges to other users.
```

=== Syntax

[source,text]
----
GRANT <privilegeName1> (, <privilegeName2>)* ON <privilegeObjects>
        [IN <privilegeScopes>]? TO <user_name1> (, <user_name2>)*
----

> === Require privilege

Reviewer 2: ```suggestion
=== Required privilege
```

`WRITE_USER`

=== Example

. To grant privileges to a user, run the `GRANT PRIVILEGE` command from the GSQL shell:
+
[source,text]
----
GSQL > GRANT WRITE ON ALL ROLES
        IN GRAPH example_graph TO user1 , user2
----

> This will allow `user1` and `user2` to modify roles on the graph `example_graph`. To see a full list of privileges and the command they allow users to run, see 

Reviewer: Be specific after this. It is not clear.

It should be like: This command allows user1 and user2 to modify roles on the example_graph. Specifically, they can perform operations that involve creating, modifying, or deleting roles within the graph.

Author: This is from our current docs in `role-management.adoc`. Whatever, I can improve it here.

> This will allow `user1` and `user2` to modify roles on the graph `example_graph`. To see a full list of privileges and the command they allow users to run, see 

Reviewer 2: ```suggestion
This command will allow `user1` and `user2` to modify roles on the graph `example_graph`.
Specifically, they can create, modify, or delete roles for that graph.
To see a full list of privileges and the command they allow users to run, see 
```

The following command grants the `READ_DATA` privilege on the `to` attribute of the edge type `Knows` to `example_user`:

[source.wrap,gsql]
----
GRANT READ ON EDGE Knows ATTRIBUTE to IN GRAPH ldbc_snb TO example_user <1>
----
<1> `to` must be lower-case.

== Revoke privileges from a user

Users with the `WRITE_USER` privileges on global can revoke RBAC privileges from the users.

=== Syntax

[source,text]
----
REVOKE <privilegeName1> (, <privilegeName2>)* ON <privilegeObjects>
        [IN <privilegeScopes>]? FROM <user_name1> (, <user_name2>)*
----

=== Required privilege

`WRITE_USER`

=== Example

. To revoke privileges from a user, run the `REVOKE PRIVILEGE` command from the GSQL shell:
+
[source,text]
----
GSQL > REVOKE WRITE ON ALL ROLES
        IN GRAPH example_graph TO user1
----

This will revoke the `WRITE_ROLE` privilege from the user `user1` on graph `example_graph.`

== Revoke type-level privileges
You can revoke certain privileges from the type level with the `REVOKE PRIVILEGE` command.

> If the privilege has already been granted on the global scope or on the graph level, then revoking the privilege on the type level does not revoke the privilege in effect.

Reviewer: This requires more clarity with an example.

Revoking a privilege at the type or attribute level does not affect privileges granted at higher levels (e.g., global or graph level). For instance, if a user has READ_DATA on a vertex type globally, revoking this privilege at the graph level will not prevent the user from accessing the vertex type elsewhere.

Author: Ditto.

Reviewer 2: @arun-TG-PM , if you are going to suggest exact wording, you might as well put it in as a suggested edit that can be accepted. Do you know how to do that?  Click the [+/-] icon, then revise the text between the quotes.
